### ISSUE TITLE
Trouble pairing my bike

### ISSUE THREAD

**Issue by pjrigatti** [cagnulein/qdomyos-zwift#2227]
**Describe the bug**
I have used this app for two years without a problem.  But for about the last two weeks I have been having issues, and can't get it to work, despite having uninstalled and reinstalled both Zwift and qz.  When I launch Zwift on my PC, I am able to pair my bike using the qz app on my iPad.  I successfully pair the Power Source, Resistance, Cadence and Heart Rate.  When I start the ride however, the Power and Cadence stay at 0, no matter how furiously I pedal.  My heart rate, which is recorded by a Wahoo Ticker device, does record data.  

![Zwift](https://github.com/cagnulein/qdomyos-zwift/assets/163468413/da277213-8b4e-4644-9a92-a69302885728)

**Desktop (please complete the following information):**
 - OS: Win 10
 - Version 22H2

**Smartphone (please complete the following information):**
 - Device: Ipad Pro 3rd Gen
 - iOS 17.4 - 

**Append a debug log**

 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
  
  
**Additional context**
[debug-Thu_Mar_14_17_31_22_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14608397/debug-Thu_Mar_14_17_31_22_2024.log)

**Comment by cagnulein** [comment]
don't you see the wahoo virtual device on zwift from qz?
why don't you use the qz version on windows too to avoid this kind of issues?

**Comment by pjrigatti** [comment]
I’m sorry, but I don’t understand.  When I first set up the qz app, I thought that I had read that qz should run on my iOS device, with Zwift on my PC.  Are you saying I should also install qz on my PC?  So I would have qz on my iPad and on my PC, both linked to Zwift?  If so, please explain how to configure this setup.  Also, when I pair my bike in Zwift, I used to see the option for both Wahoo and qz.  I have always chosen qz, because that’s what I thought was the correct virtual device.  I did not think that Wahoo would work to pair the bike.  For some reason now, Wahoo no longer shows as a device to choose.

 

I appreciate all of your help.

 

Thanks,

 

Paul

**Comment by cagnulein** [comment]
Yes 2 years ago, QZ on PC doesn't exist :) Now it does :) that's I'm telling you so now :)

Your current issue it could probably about new windows driver, windows updates, zwift udpates, there are a lot of reasons for this so that's why, in order to remove all of them in one shot, I suggest to try the QZ version on PC. If it's ok, it will be the perfect setup for you I guess.

Yes Wahoo device is the way to go, if you also lost this in Zwift, it means that something has changed on your network,

Anyway, let's give a shot to the QZ windows trial version from here: https://github.com/cagnulein/qdomyos-zwift/releases/download/nightly/windows-msvc2019-binary-no-python.zip
Just download it, unzip it on a folder and run the qdomyos-zwift.exe

If it conencts to the bike correctly and you see the metrics, open zwift on the same pc and connect it to the wahoo device.

let me know

**Comment by pjrigatti** [comment]
So I tried QZ windows.  It does connect to my bike:




 




When I launch Zwift on my app, I have the option to connect to Wahoo, but as you can see, not with the Bluetooth icon that I normally see:

 



 

I paired my bike using the Wahoo KICKR, but the same thing happened.  My pedal strokes are not transmitted to Zwift:

 



 

I hope this helps.

 

Thanks,

 

Paul

**Comment by cagnulein** [comment]
hi i don't see any images. you have to push into github directly 

anyway wahoo tickr is right

did you try to complete uninstall zwift? it sounds to me like an issue of zwift at this point.

maybe try zwift on another device too

let me know

**Comment by pjrigatti** [comment]
Hi Roberto.  I tried the complete uninstall/reinstall of zwift.  I launched QZ on my PC.  It paired with my bike.  I launched Zwift on my PC.  On the Paired Devices screen, I was able to select Wahoo KICKR 0000 33 for my Power Source, Resistance, and Cadence.  I was able to select Wahoo TICKR 6805156 for my Heart Rate, which is the heart rate monitor that I purchased.  After I clicked on OK, and chose my ride, the program launched correctly.  However, the only thing that gets transmitted to Zwift is my heart rate.  I have attached screenshots.  I also tried this on my laptop, running Windows 10, with the same results.  Thanks
![Roberto2](https://github.com/cagnulein/qdomyos-zwift/assets/163468413/96b2632f-c05c-4760-a4e2-19966b7a132d)
![Roberto1](https://github.com/cagnulein/qdomyos-zwift/assets/163468413/d2b2bdb6-5735-46e2-b025-614ada804571)
![Roberto 5](https://github.com/cagnulein/qdomyos-zwift/assets/163468413/992faac2-98ff-43d5-8a6d-0045db7130a8)
![Roberto 4](https://github.com/cagnulein/qdomyos-zwift/assets/163468413/5a8d1793-6740-4c1b-ac37-51dbf078c43e)

**Comment by cagnulein** [comment]
Ok try to use this one instead 

https://github.com/cagnulein/qdomyos-zwift/actions/runs/8308710354/artifacts/1331999042

Pair the bike first on the Bluetooth settings of windows before starting qz

Let me know

**Comment by pjrigatti** [comment]
I tried it with the same result.  I can pair the bike to my PC, then to qz, then to Zwift.  But Zwift continues to show my wattage and RPM at 0.

 

Thanks,

 

Paul

**Comment by cagnulein** [comment]
did you pair the wahoo device there? same pc  for both qz and zwift? do you
see the metrics on the qz on pc?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno mar 19 mar 2024 alle 22:41 pjrigatti ***@***.***>
ha scritto:

**Comment by pjrigatti** [comment]
![zwift 3](https://github.com/cagnulein/qdomyos-zwift/assets/163468413/469be8c8-5ee1-4f03-91fb-54d4f4fdbb79)
Yes, I paired the wahoo device.  Same PC for qz and Zwift.  I do see metrics, however the only metrics that change are the Resistance and Elapsed Time.  This has been the case since I started having issues.

**Comment by cagnulein** [comment]
ok so that's another issue! don't you have cadence also on your phone app? if so i guess you have a problem on the cadence sensor of the bike. Did you try also with the echelon app?

**Comment by pjrigatti** [comment]
Yes that is correct, there is no cadence when I connect on my iPad either.  I will try connecting to the Echelon app and let you know what happens.  Thank you.

**Comment by cagnulein** [comment]
Ok, thanks for the update.
So 99% it's an hardware issue thst can you fix buying a simple cadence
sensor and connect it to qz as well

Il giorno gio 21 mar 2024 alle 00:56 pjrigatti ***@***.***>
ha scritto:

**Comment by pjrigatti** [comment]
Hi Roberto.  I just tried my bike with the Echelon app.  You were correct, the cadence sensor is not being recognized by the app either, just the resistance.  I have a cadence sensor, by XOSS.  Is there anything special I need to do to connect it to qz?  Thanks for all of your help!

**Comment by cagnulein** [comment]
Yes just set it under accessories, cadence sensor setting and press OK and
restart qz. Let me know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 22 mar 2024 alle 02:09 pjrigatti ***@***.***>
ha scritto:

**Comment by pjrigatti** [comment]
Hey Roberto, I was able to pair my cadence sensor with qz, so I am back in business!!! Grazie di tutto!!!

**Comment by cagnulein** [comment]
glad to hear it !
